services/device-registry-base/src/main/java/org/eclipse/hono/service/management/device/DelegatingDeviceManagementHttpEndpoint.java

```diff
@@ -186,6 +186,12 @@ private void doSearchDevices(final RoutingContext ctx) {
                 RegistryManagementConstants.PARAM_FILTER_JSON, Filter.class);
         final Future<List<Sort>> sortOptions = decodeJsonFromRequestParameter(ctx,
                 RegistryManagementConstants.PARAM_SORT_JSON, Sort.class);
+        final Future<Optional<Boolean>> isGateway = getRequestParameter(
+                ctx,
+                RegistryManagementConstants.PARAM_IS_GATEWAY,
+                Optional.empty(),
+                CONVERTER_BOOLEAN,
+                value -> true);
```

Comment:
you will need to include this Future in the `CompositeFuture.all()` below to prevent a potential race condition

Reply:
Done